src/Guild.js

```diff
@@ -25,6 +25,35 @@ class Guild {
     autoBind(this);
   }
 
+  guildRoleCreate (roleId) {
+    this.roles[this.roles.name] = roleId;
+  }
+
+  guildGetKeyByValue (value) {
+    return Object.keys(this.roles).find(key => this.roles[key] === value);
+  }
+
+  guildUpdateRoleName (roleId) {
+    const oldKey = this.guildGetKeyByValue(roles.id);
+
+    //role for roleId does not exist. nothing to update
+    if(!this.isRole(oldKey)) {
+      return;
+    }
+
+    if(oldKey !== this.roles.name) {
+      this.roles[this.roles.name] = this.roles[oldKey];
+      delete this.roles[oldKey];
+    }
+
+    // given roleName is same as the one already existing - do nothing
+    return;
+  }
+
+  guildDeleteRole(roleId) {
+    delete this.roles[this.guildGetKeyByValue(roleId)];
```

Comment:
You already have the name of the Role from the event. You just need to pass it in.